customization/customizing-the-backend.md

```diff
@@ -65,59 +64,61 @@ Our new business logic is pretty easy to test in integration:
 
 {% code title="spec/models/spree/order\_spec.rb" %}
 ```ruby
+require 'rails_helper'
+require 'amazing_store/order_analyzer'
+
 RSpec.describe Spree::Order do
   describe '#finalize!' do
     before do
-      allow(ENV).to receive(:fetch)
-        .with('BLACKLISTED_EMAILS', any_args)
-        .and_return('jdoe@example.com')
-      end
+    stub_const 'ENV',
+        ENV.to_h
+        .merge('REJECTED_EMAILS' => 'jdoe@example.com')
     end
 
-    context 'when the email has been blacklisted' do
-      it 'marks the order as blacklisted' do
-        order = create(:order_ready_to_complete, email: 'jdoe@example.com')
-
+    context 'when the email has been rejected' do
+      it 'marks the order as rejected' do
+        order = create(:order_ready_to_complete)
+        order.update(email: 'jdoe@example.com')
         order.finalize!
 
-        expect(order).to be_blacklisted
+        expect(order).to be_rejected
       end
     end
 
-    context 'when the email has not been blacklisted' do
-      it 'does not mark the order as blacklisted' do
-        order = create(:order_ready_to_complete, email: 'hello@example.com')
-
+    context 'when the email has not been rejected' do
+      it 'does not mark the order as rejected' do
+        order = create(:order_ready_to_complete)
+        order.update(email: 'hello@example.com')
         order.finalize!
 
-        expect(order).not_to be_blacklisted
+        expect(order).not_to be_rejected
       end
     end
   end
 end
 ```
 {% endcode %}
 
-At this point, we have a dead-simple order analyzer that determines whether each new order should be blacklisted or not. Now, we need to allow admins to manually review blacklisted orders and decide whether to reject them or remove them from the blacklist.
+At this point, we have a dead-simple order analyzer that determines whether each new order should be rejected or not. Now, we need to allow admins to manually review rejected orders and decide whether to reject them or remove them from the rejected.
 
 ## Implementing new actions
 
-In order to allow admins to remove an order from the blacklist, we'll add a button to the order detail page that will trigger a new controller action.
+In order to allow admins to remove an order from the rejected, we'll add a button to the order detail page that will trigger a new controller action.
 
 The first step is to add our custom action to `Spree::Admin::OrdersController`. We'll use a decorator to accomplish that:
 
-{% code title="app/decorators/awesome\_store/spree/admin/oders\_controller/add\_remove\_from\_blacklist\_action.rb" %}
+{% code title="app/decorators/amazing\_store/spree/admin/orders\_controller/add\_remove\_from\_rejected\_action\_decorator.rb" %}
```

Comment:
Why is `_decorator` added here?

Reply:
On a new solidus application install, the installer sets up the configuration to look for decorators using this syntax:

config/application.rb
` 
...
initializer 'spree.decorators' do |app|
    config.to_prepare do
       Dir.glob(Rails.root.join('app/**/*_decorator*.rb')) do |path|
          require_dependency(path)
       end
   end
end
...
`
perhaps it was intended as:
`
initializer 'spree.decorators' do |app|
    config.to_prepare do
       Dir.glob(Rails.root.join('app/decorators/**/*.rb')) do |path|
          require_dependency(path)
       end
   end
end
`
What do you think?

Reply:
Yes, I guess we should change that in core. Want to submit a PR for that after trying if that works in your application?

Reply:
submitted  [#4054](https://github.com/solidusio/solidus/pull/4054)